packages/form/src/FormAutocomplete/FormAutocomplete.test.tsx

```diff
@@ -219,4 +221,55 @@ describe('FormAutocomplete', () => {
       expect(onInputChange.mock.calls).toHaveLength(4);
     });
   });
+
+  it('Value изменяется при изменении значения формы', async () => {
+    type FormFreeValues = {
+      autocomplete: Option | string;
+    };
+
+    // eslint-disable-next-line quotes
+    const formText = "It's sunday!";
+
+    const TestComponent = () => {
+      const [, setIsUpdate] = useState(false);
+
+      const form = useForm<FormFreeValues>({
+        reValidateMode: 'onChange',
+      });
+
+      const handleButtonClick = () => {
+        form.setValue('autocomplete', formText);
+        setIsUpdate(true);
+      };
+
+      return (
+        <Form form={form}>
+          <FormAutocomplete<FormFreeValues, Option, true, false, true>
+            control={form.control}
+            name="autocomplete"
+            options={[]}
+            freeSolo
+            label="Form autocomplete with freeSolo"
+            getOptionLabel={(params) =>
+              typeof params === 'string' ? params : params.name
+            }
+          />
+
+          <Button id="set-text-btn" onClick={handleButtonClick}>
+            Set text
+          </Button>
+        </Form>
+      );
+    };
+
+    renderWithTheme(<TestComponent />);
+
+    const setTextBtn = screen.getByRole('button');
+
+    userEvents.click(setTextBtn);
```

Comment:
```suggestion
    await userEvents.click(setTextBtn);
```

waitFor думаю не понадобится после этого